### PR TITLE
Development

### PR DESCRIPTION
**Pull request by diegoarvz4**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+          redirect_to new_user_path
+        else
+          render :new
+        end
+    end
+
+    def edit
+        @user = User.find(params[:id])
+    end
+    def update
```
[the remark this answers is not in the document]
Comment (dipto0321) [reply app/controllers/users_controller.rb]: Please complete this method

File: app/views/users/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    <%= render 'shared/error_messages'%>
+    
+    <%= t.label :username %>
+    <%= t.text_field :username %>
+    <%= t.label :email%>
+    <%= t.text_field :email ,placeholder: "diego@diego.com" %>
+    <%= t.label :password %>
+    <%= t.password_field :password %>
+
+    <%= t.submit :Register %>
```
Comment (dipto0321) [inline app/views/users/new.html.erb]: No need `:Register`

File: config/routes.rb [hunk 3]
```diff
@@ -1,3 +1,4 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+ resources :users
```
Comment (dipto0321) [inline config/routes.rb]: This project only can edit, create, update and new. So we can also do this `resources :users, only: [:new, :create, :edit, :update]`

Comment (diegoarvz4) [reply config/routes.rb]: Ok. But as your first directive says " it is possible to edit a user in app this is a very important feature for this project", we needed to add :index in the route. so we can have a view to edit a specific user in /users.

Comment (dipto0321) [reply config/routes.rb]: Okay. That time I saw you just use new, create, edit, update. That's why I just share this to you. It's completely fine if you stick with the previous one.

File: config/routes.rb [hunk 4]
```diff
@@ -1,4 +1,4 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
- resources :users
+  resources :users, only: [:index, :new, :create, :edit, :update]
```
Comment (dipto0321) [inline config/routes.rb]: By the way, you don't need to change anything here. Just sharing this tip with you. There is 7 operation in rails. index, new, create, show, edit, update, destroy. When we need all of these then we use simple `resources` and the name of the controller. If we need only couple of them as a route then we use `only:`. If we saw that we use the maximum of those as routs except one or two items from controller then we can use `except:` . So we can also write this like this ` resources :users, except: [:show, :destroy]`. You will get it [here](https://guides.rubyonrails.org/routing.html#restricting-the-routes-created).

File: config/routes.rb (hunk 3 above)
Comment (diegoarvz4) [reply config/routes.rb]: Cool. Thank you very much Dipto !

**Pull request merged**